client/benchmark/src/main/java/org/elasticsearch/client/benchmark/rest/RestClientBenchmark.java

```diff
@@ -0,0 +1,136 @@
+ * under the License.
+ */
+package org.elasticsearch.client.benchmark.rest;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpHost;
+import org.apache.http.HttpStatus;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.client.StandardHttpRequestRetryHandler;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.benchmark.AbstractBenchmark;
+import org.elasticsearch.client.benchmark.ops.bulk.BulkRequestExecutor;
+import org.elasticsearch.client.benchmark.ops.search.SearchRequestExecutor;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+
+public final class RestClientBenchmark extends AbstractBenchmark<RestClient> {
+    public static void main(String[] args) throws Exception {
+        RestClientBenchmark b = new RestClientBenchmark();
+        b.run(args);
+    }
+
+    @Override
+    protected RestClient client(String benchmarkTargetHost) {
+        CloseableHttpClient httpClient = HttpClients
+            .custom()
+            .setRetryHandler(new StandardHttpRequestRetryHandler(3, true))
+            .build();
+
+        return RestClient.builder(new HttpHost(benchmarkTargetHost, 9200))
+            .setHttpClient(httpClient)
+            .build();
+    }
+
+    @Override
+    protected BulkRequestExecutor bulkRequestExecutor(RestClient client, String indexName, String typeName) {
+        return new RestBulkRequestExecutor(client, indexName, typeName);
+    }
+
+    @Override
+    protected SearchRequestExecutor searchRequestExecutor(RestClient client, String indexName) {
+        return new RestSearchRequestExecutor(client, indexName);
+    }
+
+    private static final class RestBulkRequestExecutor implements BulkRequestExecutor {
+        private final RestClient client;
+        private final String actionMetaData;
+
+        public RestBulkRequestExecutor(RestClient client, String index, String type) {
+            this.client = client;
+            this.actionMetaData = String.format(Locale.ROOT, "{ \"index\" : { \"_index\" : \"%s\", \"_type\" : \"%s\" } }%n", index, type);
+        }
+
+        @Override
+        public boolean bulkIndex(List<String> bulkData) {
+            StringBuilder bulkRequestBody = new StringBuilder();
+            for (String bulkItem : bulkData) {
+                bulkRequestBody.append(actionMetaData);
+                bulkRequestBody.append(bulkItem);
+                bulkRequestBody.append("\n");
+            }
+            StringEntity entity = new StringEntity(bulkRequestBody.toString(), ContentType.APPLICATION_JSON);
+            Response response = null;
+            try {
```

Comment:
oh you did this because of the IOException thrown by close....in that case leave it. it will change soon anyway :)